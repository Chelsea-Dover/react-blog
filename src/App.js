import React, { Component } from 'react'
import './client/App.css'
import Header from './Header'
import Footer from './Footer'
import AllArticles from './AllArticles'
import SeparatePost from './SeparatePost'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import marked from 'marked'

const SERVER_ROOT = 'https://cmsblog-aaa94.firebaseio.com'

/* global fetch */

class App extends Component {
  constructor (props) {
    super(props)
    this.state = {
      markdownList: [],
      config: [],
      contact: [],
      titles: []
    }
  }
  componentWillMount () {
    fetch(`${SERVER_ROOT}/articles.json`)
      .then(response => {
        console.log('It worked!')
        console.log(response)
        return response.json()
      })
      .then(text => {
        const newMarkdownList = this.state.markdownList.map((item) => Object.assign({}, item))
        for (const x in text) {
          newMarkdownList.push({content: marked(text[x].data), path: x})
          this.setState({
            markdownList: newMarkdownList
          })
        }
      })

    fetch(`${SERVER_ROOT}/userConfig/config.json`)
      .then(output => {
        console.log('It worked!')
        return output.json()
      })
      .then((userConfig) => {
        this.setState({config: userConfig.config})
        this.setState({contact: userConfig.contact})
        this.setState({titles: userConfig.titles})
      })
  }

  render () {
    const SinglePost = ({ match }) => {
      if (match === undefined) {
        return null
      }
      return (
        <SeparatePost
          articleName={match.params.id}
          markdownList={this.state.markdownList}
        />
      )
    }
    const ArticleList = () => (
      <AllArticles
        markdownList={this.state.markdownList}
      />
    )
    return (
      <Router>
        <div className={this.state.config.blogTheme}>
          <Header
            blogTitle={this.state.titles.blogTitle}
            subTitle={this.state.titles.subTitle}
          />
          <Route path='/post/:id' component={SinglePost} />
          <Route exact path='/' component={ArticleList} />
          <Footer
            footerMessage={this.state.titles.footerMessage}
          />
        </div>
      </Router>
    )
  }
}

export default App
