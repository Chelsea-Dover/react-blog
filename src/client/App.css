/* If your css doesn't override the default try using the :important rule */
/* styles here */
