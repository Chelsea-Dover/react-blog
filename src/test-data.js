export const FAKE_DATA = [
  { content: '<h1>Hello!</h1>', path: 'pathHello' },
  { content: '<h2>Hey!</h2>', path: 'pathHey' },
  { content: '<h2>Howdy!</h2>', path: 'pathHowdy' }
]
