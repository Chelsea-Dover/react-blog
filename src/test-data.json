{
  "articles":[
  { "pathHello": "# Hello!" },
  { "pathHey": "# Hey!" },
  { "pathHowdy": "#Howdy!" }
]
}
