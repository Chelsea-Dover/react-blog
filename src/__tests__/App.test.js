import React from 'react'
import App from '../App'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import fetch from 'jest-fetch-mock'
import {FAKE_DATA} from '../test-data'

global.fetch = fetch

/* global expect describe it beforeEach */

fetch.mockResponse(JSON.stringify(FAKE_DATA))

describe('App tests', () => {
  it('does a snapshot check', () => {
    console.log(FAKE_DATA)
    const component = renderer.create(<App />)
    const json = component.toJSON()
    expect(json).toMatchSnapshot()
  })
  describe('', () => {
    let wrapper
    beforeEach(() => {
      wrapper = shallow(
        <App
        />
      )
    })
    it('', () => {
      console.log(wrapper.props())
    })
  })
})
