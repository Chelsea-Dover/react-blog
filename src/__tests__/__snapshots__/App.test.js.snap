// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests does a snapshot check 1`] = `
<div
  className={undefined}
>
  <div
    className="headerWrapper"
  >
    <img
      alt="undefined logo"
      src="logo.jpg"
      style={
        Object {
          "border": "solid 2px #eee",
          "borderRadius": "50%",
          "height": "6em",
          "width": "6em",
        }
      }
    />
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        <h1
          className="headerLink"
        />
      </a>
      <h2
        className="subheadStyle"
      />
    </div>
  </div>
  <div
    className="AllArticleWrapper"
    style={
      Object {
        "paddingBottom": "140px",
      }
    }
  />
  <div
    className="footerWrapper"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "padding": "20px 100px",
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <h2 />
  </div>
</div>
`;
